packages/server/src/supertokens.ts

```diff
@@ -723,7 +719,7 @@ export async function createNewSession(
 }
 
 export async function createAnonymousSession(req: IncomingMessage, res: ServerResponse) {
-  return await createNewSession(req, res, {userId: null, roles: []}, undefined, {anonymous: true})
+  return await createNewSession(req, res, {userId: null}, undefined, {anonymous: true})
```

Comment:
I think we need to add a "default publicData" config that would be used here for anonymous sessions. Because with as setup where `roles` is used, you always want `roles` to be an array type.